gdk4/src/auto/event.rs

```diff
@@ -0,0 +1,205 @@
+    pub fn device_tool(&self) -> Option<DeviceTool> {
+        unsafe {
+            from_glib_none(ffi::gdk_event_get_device_tool(
+                self.as_ref().to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "gdk_event_get_display")]
+    #[doc(alias = "get_display")]
+    pub fn display(&self) -> Option<Display> {
+        unsafe { from_glib_none(ffi::gdk_event_get_display(self.as_ref().to_glib_none().0)) }
+    }
+
+    #[doc(alias = "gdk_event_get_event_sequence")]
+    #[doc(alias = "get_event_sequence")]
+    pub fn event_sequence(&self) -> Option<EventSequence> {
+        unsafe {
+            from_glib_none(ffi::gdk_event_get_event_sequence(
+                self.as_ref().to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "gdk_event_get_event_type")]
+    #[doc(alias = "get_event_type")]
+    pub fn event_type(&self) -> EventType {
+        unsafe {
+            from_glib(ffi::gdk_event_get_event_type(
+                self.as_ref().to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "gdk_event_get_history")]
+    #[doc(alias = "get_history")]
+    pub fn history(&self) -> Vec<TimeCoord> {
+        unsafe {
+            let mut out_n_coords = mem::MaybeUninit::uninit();
+            let ret = FromGlibContainer::from_glib_container_num(
+                ffi::gdk_event_get_history(
+                    self.as_ref().to_glib_none().0,
+                    out_n_coords.as_mut_ptr(),
+                ),
+                out_n_coords.assume_init() as usize,
+            );
+            ret
+        }
+    }
+
+    #[doc(alias = "gdk_event_get_modifier_state")]
+    #[doc(alias = "get_modifier_state")]
+    pub fn modifier_state(&self) -> ModifierType {
+        unsafe {
+            from_glib(ffi::gdk_event_get_modifier_state(
+                self.as_ref().to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "gdk_event_get_pointer_emulated")]
+    #[doc(alias = "get_pointer_emulated")]
+    pub fn is_pointer_emulated(&self) -> bool {
+        unsafe {
+            from_glib(ffi::gdk_event_get_pointer_emulated(
+                self.as_ref().to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "gdk_event_get_position")]
+    #[doc(alias = "get_position")]
+    pub fn position(&self) -> Option<(f64, f64)> {
+        unsafe {
+            let mut x = mem::MaybeUninit::uninit();
+            let mut y = mem::MaybeUninit::uninit();
+            let ret = from_glib(ffi::gdk_event_get_position(
+                self.as_ref().to_glib_none().0,
+                x.as_mut_ptr(),
+                y.as_mut_ptr(),
+            ));
+            let x = x.assume_init();
+            let y = y.assume_init();
+            if ret {
+                Some((x, y))
+            } else {
+                None
+            }
+        }
+    }
+
+    #[doc(alias = "gdk_event_get_seat")]
+    #[doc(alias = "get_seat")]
+    pub fn seat(&self) -> Option<Seat> {
```

Comment:
This is not `nullable`, is it? Probably similar problem for other functions

Reply:
Yes, see my latest comment. They were nullable in the manual bindings as well